tests/templates/test_layers/test_strongly_entangling.py

```diff
@@ -89,6 +89,30 @@ def circuit2():
 
         assert np.allclose(dev.state, dev2.state, atol=tol, rtol=0)
 
+    @pytest.mark.parametrize(
+        "n_layers, n_wires, ranges", [(2, 2, [1]), (1, 3, [2]), (4, 4, [2, 3, 1])]
+    )
+    def test_custom_range_sequence(self, n_layers, n_wires, ranges):
+        """Test that correct sequence of custom ranges are used in the circuit."""
+
+        weights = np.random.randn(n_layers, n_wires, 3)
+
+        op = qml.templates.StronglyEntanglingLayers(
+            weights=weights, wires=range(n_wires), ranges=ranges
+        )
+        ops = op.expand().operations
+
+        gate_wires = [gate.wires for gate in ops]
```

Comment:
You could extract `gate.wires.labels` here, and then use a tuple for the `test_wire`, so you don't need the `Wires` class below...